assignments/nested-parens.adoc

```diff
@@ -0,0 +1,228 @@
+= Exercise: Nested Parens
+:source-language: rust
+
+In this exercise, you will learn
+
+* how to translate a `for loop` into an `iterator` solution by using `map`, `scan`, and `fold`
+* how to setup a benchmarking harness to compare speeds
+* how to use `Rayon` to parallelize your iterators!
+* how to compress data to get even more speedups
+
+We will take a famous coding problem and iterate solutions, and measure their performance.
```

Comment:
I'm getting advent of code vibes :smile:

Reply:
Yes, it's a classic problem that a lot of Competitive coding places use / algorithms class also use